### PR TITLE
UI: Tidy up login and homepage

### PR DESCRIPTION
**Pull request by whyman**
New login view
grb-login
New homepage layout
grb-home-with-content
New homepage without content
grb-home
Homepage reflows on small screens
grb-home-small

### CODE REVIEW

**Comment by whyman** [comment]
Havent touched the tests, so they will probably need work.

**Comment by KarlStraussberger** [comment]
Seems there is one problem

[1]        The login action
[1]          "before each" hook for "hides the login form when no authentication is required":
[1]      TimeoutError: Waiting until element is visible
[1] Wait timed out after 5174ms
[1]       at /home/runner/work/gerbera/gerbera/gerbera-web/node_modules/selenium-webdriver/lib/webdriver.js:929:17
[1]       at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

**Comment by KarlStraussberger** [comment]
The test requires an input with `login-fields` style (it a property in login.page.js)

**Comment by KarlStraussberger** [comment]
I'll see to create the new screenshots for the docs.

File: web/js/gerbera-auth.module.js [hunk 1]
```diff
@@ -125,6 +125,7 @@ const submitLogin = (response) => {
 
 const checkLogin = (response) => {
   if (response.success) {
+    $('#login').hide();
     $('.login-field').hide();
```
Comment (KarlStraussberger) [inline web/js/gerbera-auth.module.js:129]: Maybe that's causing an error

File: web/js/gerbera-app.module.js [hunk 2]
```diff
@@ -343,7 +343,9 @@ export class App {
   displayLogin (loggedIn) {
     $('#server-status').hide();
     $('#server-empty').hide();
+    $('#homelinks').hide();
     if (loggedIn) {
+      $('#login').hide();
       $('.login-field').hide();
```
Comment (KarlStraussberger) [inline web/js/gerbera-app.module.js:349]: That's another one

File: web/index.html [hunk 3]
```diff
@@ -47,156 +47,197 @@
...
         </div>
       </div>
-    </div>
-    <div class="row mx-2" id="homelinks" style="display: none">
-      <div class="col md-3 offset-md-1">
-        <a id="addlink" href="javascript:;" onclick="$('#nav-fs').click(); // Replace with hashchange handling">
-          <div class="sublink">
-            <h4><i class="fa fa-plus"></i> Add some files</h4>
-            <span>Browse and import your multimedia files into Gerbera.</span>
+      <div id="login">
```
Comment (KarlStraussberger) [inline web/index.html:68]: Maybe it help to hide it with `style="display:none"`

**Review by KarlStraussberger — COMMENTED** [review]
Just some ideas from my _mobile debugging session_

**Comment by udippel** [comment]
Just asking: Have you considered some method (middleware?) to follow up on token expiration? 
At least to me that's the item that bothers me most of all. Clicking some item only to be told that the session has expired and then having to go back manually to me is much worse than at least stating automatically that the session has expired; better still: to pop up the login screen automatically after expiration.

**Comment by whyman** [comment]
:( tests are passing fine on my machine, frustrating

**Comment by KarlStraussberger** [comment]
My record of reruns was well above 10 :hooray:

**Pull request merged**